introspection/org.freedesktop.NetworkManager.Device.Wireguard.xml

```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<node name="/">
+  <!--
+      org.freedesktop.NetworkManager.Device.Wireguard:
+      @short_description: WireGuard Device
+
+  -->
+  <interface name="org.freedesktop.NetworkManager.Device.Wireguard">
+
+    <!--
+        ListenPort:
+
+        Local UDP listening port.
+    -->
+    <property name="ListenPort" type="u" access="read"/>
+
+    <!--
+        Fwmark:
+
+        Firewall mark.
+    -->
+    <property name="Fwmark" type="u" access="read"/>
+
+    <!--
+        PrivateKey:
+
+        32-byte private WireGuard key.
+    -->
+    <property name="PrivateKey" type="v" access="read"/>
+
+    <!--
+        PublicKey:
+
+        32-byte public WireGuard key.
+    -->
+    <property name="PublicKey" type="v" access="read"/>
+
```

Comment:
Please include a "Parent" property too.

Reply:
This is a question I meant to ask too - what is the purpose of the parent device?

Reply:
On the second thought -- I have no idea if this is relevant for Wireguard. Maybe not.

Links that are associated with some other interface have this. E.g. VLANs, Tunneling devices or Veth pairs. In general it mirrors the IFLA_LINK property of the rtnetlink link. (In other words, if it says "link1@link0:" in "ip link", then link0 is the parent device.)

Reply:
In that case, I don't think it's relevant, no. Wireguard is layer 3, it doesn't care about another particular link.